bundler/lib/bundler/definition.rb

```diff
@@ -287,50 +286,6 @@ def resolve
       end
     end
 
-    def index
```

Comment:
@deivid-rodriguez, it seems that the removal of this method broke https://rubygems.org/gems/gemsurance as that method was used [here](https://github.com/appfolio/gemsurance/blob/2680f874c76247cf41fface3656b7764e817457c/lib/gemsurance/gem_info_retriever.rb#L122), resulting in a `NoMethodError` on bundler v2.2.18:
```
undefined method `index' for #<Bundler::Definition:0x00007fd7dd23a958> (NoMethodError)
```

Given that this was removed in a patch version of bundler (the method existed in bundler v2.2.17), I'm guessing it wasn't considered public API in the first place. But I figured I'd mention it in case it was okay to use the `index` method externally and this breakage was actually unintended.

Reply:
Hei @rmacklin, thanks for letting me know.

Yeah, this is not public API. The method was using a merged index (gems coming from different sources being used indifferently), which was the underlying problem causing "dependency confusion". So I decided to remove it since I think it was making `bundle outdated` suggest unexpected (potentially insecure) gems.

I guess `gemasurance` can adapt https://github.com/rubygems/rubygems/pull/4609/commits/1d1d8a8f0816c9734cd0500da49df68cfa50bdc5 to its needs: use the same source as the currently resolved spec instead of using a merged source (note that commit caused a regression fixed here: https://github.com/rubygems/rubygems/commit/64168ebceb6d99b27c189ad7ec4819c9da69f502.

Reply:
Thanks for the confirmation!

I think we may be able to replace `gemsurance` with a combination of `bundler-audit` and `bundle outdated`.